pkg/controller/job/controller.go

```diff
@@ -403,10 +398,6 @@ func (jm *JobManager) manageJob(activePods []*api.Pod, successful, unsuccessful
 	} else if active < parallelism {
 		// how many executions are left to run
 		diff := *job.Spec.Completions - successful
-		// for RestartPolicyNever we need to count unsuccessful pods as well
```

Comment:
Thanks, I'll move entire discussion there. Besides thanks for all the fixes you've made in the mean time to the jobs!